src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocumentElement.cs

```diff
@@ -365,6 +366,21 @@ static string ToString (string value, ICustomAttributeProvider provider, IAssemb
 			return ToString (typeDef);
 		}
 
+		static string ToString (ForegroundService value)
+		{
+			switch (value) {
+				case ForegroundService.TypeNone: 		return "none";
+				case ForegroundService.TypeConnectedDevice:	return "connectedDevice";
+				case ForegroundService.TypeDataSync:		return "dataSync";
+				case ForegroundService.TypeLocation:		return "location";
+				case ForegroundService.TypeManifest:		return "manifest";
```

Comment:
Oddly, `manifest` isn't a documented value at: https://developer.android.com/guide/topics/manifest/service-element

I'm not sure what to make of that.  Perhaps it shouldn't be supported, and should hit the `default` case?

Reply:
Good catch!
I opened Android Studio and it is not an option on the given list. So I guess it is not supported and should hit default case.

Reply:
This is something to keep in mind if/when you get around to auto-producing these custom attributes based on the manifest XML: the XML may contain values which are not documented and which Android Studio doesn't "support".

I'm not sure if this can be fully automated.

(Though even partial automation would presumably be an improvement.)